doc/contribute/guidelines.rst

```diff
@@ -1027,8 +1027,26 @@ Note that adding a new version of a widely used API while maintaining
 support for the old one is not a treewide change. Deprecation and removal of
 such APIs, however, are treewide changes.
 
+Push CI testing
+===============
+
+Pull requests CI runs on a different testplan than what is used by the push CI
+run. For large changes, such as treewide ones, this means it's likely to
+introduce some regression that does not get caught by the PR run, but gets
+detected by the push run once merged.
+
+For this reason, it's often useful to do a separate CI run using the push
+configuration. This can be done by manually pushing the PR in the
+``zephyr-testing`` repository, for example by using:
```

Comment:
Yeah fair enough, sorry I failed to make it clear in the proposed text but the target I had in mind for that paragraph was the release team, as in "before merging a treewide consider pushing it in testing" or "ask someone for release to run it for you". At least that's something I do regularly for my patches and I pushed some for others as well, and ultimately that saved plenty of CI breakages.

But yeah ideally this would trigger automatically with the treewide tag or something more refined, was half thinking on how to implement that but I guess it probably needs some refactoring  of the workflow, or maybe some logic to switch from the PR to the push run.

Reply:
But at least it seems like we agree that we needs something there, every treewide ends up breaking the push CI and then we start screaming "revert" and the author has to fix it in hurry etc etc. Not ideal.

Reply:
Simple label-based triggering of a push-equivalent run can be prone to abuse because anyone with just triage permission level (i.e. all collaborators and maintainers) can add the label and trigger the _expensive_ push-equivalent run on any PR.

I will give this some more thought and see if there is a _safe_ way of implementing this using the current GitHub feature limitations.